### ISSUE TITLE
User Profiles use login as id

### ISSUE THREAD

**Issue by mnquintana** [YaleSTC/shifts#353]
The `/user_profiles/:id` URL uses the parameter `login` (ie. NetID) as the id, instead of the user's actual id.

So you get URLs like:

`localhost:3000/user_profiles/dsw44`

when you should be getting:

`localhost/user_profiles/102` where 102 is the user's id.

It's likely that we had this in place as a bootleg workaround for not being able to search user profiles by NetID (or by anything at all - there is no search). Instead of an admin having to wonder what Shifts user id maps to which NetID, they could just type in the NetID in the URL.

~~I think the best way to handle this is to put in a redirect for the NetID, and redirect it to the id.~~

JK, I don't think there's an elegant solution for aliasing a NetID in the URL to the User id. No matter what is after `user_profiles/`, it's assigned to `params[:id]`. In other words, we can't rely on the key name to tell us whether it's a `login` or a `user_id` - both are passed as `:id`. So we'd have to basically check to see if the param looks like a valid id (ie. if it's a positive integer), and if it doesn't, to initiate a search for the string to see if it matches a `login`. This is something we'd have to do on _every_ request, and it would only get slower with more users in the database. 

So in other words, we're much better off following convention and just having the `:id` param map to `user_id`, and adding a separate user_profiles search feature.
